cdc/redo/reader/file.go

```diff
@@ -121,134 +113,99 @@ func newReader(ctx context.Context, cfg *readerConfig) ([]fileReader, error) {
 	return readers, nil
 }
 
-func selectDownLoadFile(
-	ctx context.Context, extStorage storage.ExternalStorage, fixedType string,
-) ([]string, error) {
-	files := []string{}
-	err := extStorage.WalkDir(ctx, &storage.WalkOption{},
-		func(path string, size int64) error {
-			fileName := filepath.Base(path)
-			_, fileType, err := redo.ParseLogFileName(fileName)
-			if err != nil {
-				return err
-			}
-
-			if fileType == fixedType {
-				files = append(files, path)
-			}
-			return nil
-		})
+func downLoadAndSortFiles(ctx context.Context, cfg *readerConfig) ([]io.ReadCloser, error) {
+	dir := cfg.dir
+	// create temp dir in local storage
+	err := os.MkdirAll(dir, redo.DefaultDirMode)
 	if err != nil {
-		return nil, cerror.WrapError(cerror.ErrExternalStorageAPI, err)
+		return nil, cerror.WrapError(cerror.ErrRedoFileOp, err)
 	}
 
-	return files, nil
-}
-
-func downLoadToLocal(
-	ctx context.Context, dir string,
-	extStorage storage.ExternalStorage,
-	fixedType string, workerNum int,
-) error {
-	files, err := selectDownLoadFile(ctx, extStorage, fixedType)
+	// get all files
+	extStorage, err := redo.InitExternalStorage(ctx, cfg.uri)
 	if err != nil {
-		return err
+		return nil, err
+	}
+	files, err := selectDownLoadFile(ctx, extStorage, cfg.fileType, cfg.startTs)
+	if err != nil {
+		return nil, err
 	}
 
-	limit := make(chan struct{}, workerNum)
+	limit := make(chan struct{}, cfg.workerNums)
 	eg, eCtx := errgroup.WithContext(ctx)
+	sortedFileNames := make([]string, 0, len(files))
 	for _, file := range files {
 		select {
-		case <-ctx.Done():
-			return ctx.Err()
+		case <-eCtx.Done():
+			return nil, eCtx.Err()
 		case limit <- struct{}{}:
 		}
-		f := file
+
+		fileName := file
```

Comment:
It's better to comment why we need `fileName` instead of using `file` directly.

Reply:
Since we need to use this variable in the closure:
![image](https://user-images.githubusercontent.com/61726649/229044921-6b7fa89f-562e-4951-a608-2070970adcd1.png)